packages/app/src/vis-packs/core/matrix/MappedMatrixVis.tsx

```diff
@@ -38,14 +42,18 @@ function MappedMatrixVis(props: Props) {
     <>
       {toolbarContainer &&
         createPortal(
-          <MatrixToolbar dataset={dataset} selection={selection} />,
+          <MatrixToolbar
+            dataset={dataset}
+            selection={selection}
+            cellWidth={cellWidth}
+          />,
           toolbarContainer
         )}
 
       <MatrixVis
         dataArray={mappedArray}
         formatter={formatter}
-        cellWidth={cellWidth}
+        cellWidth={customCellWidth ?? cellWidth}
```

Comment:
Same concept as the custom domain: if `undefined`, use the default. Also, we pass the default value to the toolbar so we can show it as placeholder and check the user's input against it.